src/util/std_expr.h

```diff
@@ -824,6 +824,51 @@ inline multi_ary_exprt &to_multi_ary_expr(exprt &expr)
   return static_cast<multi_ary_exprt &>(expr);
 }
 
+/// \brief A base class for int-range expressions
+class range_exprt : public expr_protectedt
+{
+public:
+  range_exprt(exprt &lower, exprt &upper, typet _type)
+    : expr_protectedt(
+        ID_range,
+        std::move(_type),
+        {std::move(lower), std::move(upper)})
+  {
+  }
+
+  // make op0 and op1 public
+  using exprt::op0;
```

Comment:
*strong* preference to give these sensible names. You already call them "lower" and "upper", what's wrong with just using these as accessors instead of the opaque "op0" and "op1"?

Reply:
Implemented as suggested.